### ISSUE TITLE
nextflow fails with misleading error if 'mkfifo' fails

### ISSUE THREAD

**Issue by Wildish-LBL** [nextflow-io/nextflow#213]
Summary:
mkfifo can fail on NFS-mounted filesystems or other shared filesystems that do not support that operation. If nextflow fails because of this, it complains that the user-process exited with error, instead of warning the user that they should redirect the work directory with -w or NXF_WORK

How to reproduce:
1) I use a Vagrant box, running out of /vagrant, which is mounted from the host OS. Doubtless other network-mounted filesystems (e.g. Samba) would show the same problem.

2) Use a simple nextflow script:

```
$ cat /vagrant/fail.nf 
#!/home/vagrant/bin/nextflow

process sayHello {
    """
    echo 'Hello world!'
    """
}
```

3) run normally:

```
$ cd /vagrant
$ ./fail.nf 
N E X T F L O W  ~  version 0.21.1
Launching ./fail.nf
[warm up] executor > local
[bc/6d3788] Submitted process > sayHello (1)
Error executing process > 'sayHello (1)'

Caused by:
  Process 'sayHello (1)' terminated with an error exit status

Command executed:

  echo 'Hello world!'

Command exit status:
  1

Command output:
  (empty)

Work dir:
  /vagrant/jgi-workflows/new-tools/nextflow/demo/work/bc/6d3788ffe7c6e7e55c2ff02ec1dc73

Tip: view the complete command output by changing to the process work dir and entering the command: 'cat .command.out'
```

4) set the work directory to /tmp, run again, it works:

```
$ NXF_WORK=/tmp ./fail.nf 
N E X T F L O W  ~  version 0.21.1
Launching ./fail.nf
[warm up] executor > local
[a9/6d03d2] Submitted process > sayHello (1)
```

5) suggested fix:
trap errors from mkfifo and warn the user to redirect their 'work' directory. That would have saved me some time yesterday :-)

**Comment by pditommaso** [comment]
I'm sorry that you a wasted some time about. I agree that the this error message must be improved. However, as far as I know, this `mkfifo` problem is related only to the NFS client provided by Vagrant.

**Comment by Wildish-LBL** [comment]
it's not Vagrant-specific. FIFO's are supposed to be an IPC between processes on the same node, and having them communicate via a network-based filesystem can easily break that, because of buffering etc.

Either way, it's not nice to point the user in the wrong direction when it occurs :-)

**Comment by pditommaso** [comment]
We are using nextflow for something like three years over NFS with very large workloads and we never experienced this problem. The same for users deploying it over Lustre, GPFS, BeeFs. We have also users at JGI (as the path in your log seems suggesting) and they never reported such problem. 

Are you experiencing this issue in a Vagrant VM only or also in a real production system?

**Comment by Wildish-LBL** [comment]
Hi Paolo,

  I've just started exploring nextflow, for the JGI as you surmised. 
I've not deployed it on any production system yet, I see this problem 
only in my development system, so far.

  Nonetheless, the problem exists. If you choose not to fix it, that's 
fine, at least you're aware of it now.

  Tony.

**Comment by pditommaso** [comment]
Hi Tony, we take very seriously users feedback and I really thank you for opening this issue. 

I only needed to understand exactly in which environment you were experiencing this problem for troubleshooting purpose and to assign the right priority to it.

**Comment by pditommaso** [comment]
One more thing: you can set the [scratch](https://www.nextflow.io/docs/latest/process.html#scratch) directive to `true` to avoid this problem. For example: 

    nextflow run hello -process.scratch true

**Comment by pditommaso** [comment]
Included in version `0.22.4`
